### PR TITLE
#1571 - Create vuex module for chat

### LINKED ISSUE #1571: Create Vuex module for chat related settings
### Problem

The `state.js` file is quite large still and we are slowly improving its organization. We've already moved a lot of state & mutations & getters into two separate vuex modules: `'~/frontend/model/notifications/vuexModule.js'` & `'~/frontend/model/settings/vuexModule.js'`, but the chat related settings are still in `state.js`

### Solution

Similar to how the notifications and settings vuex modules work, create a `model/chatroom/vuexModule.js` file & folder and move the chatroom related state/mutations/getters into that and update all dependent files.

### PR DESCRIPTION
**Pull request by SebinSong**
closes #1571 

The work was done based on the docs here:  https://v3.vuex.vuejs.org/guide/modules.html

### CODE REVIEW

File: frontend/model/state.js
Comment (taoeffect) [inline frontend/model/state.js]: I think we can delete this whole section now since the groups will need to be recreated anyway.

**Review by taoeffect — DISMISSED** [review]
Nice! I couldn't find any issues... but this is a significant change so I'm going to wait on either @Silver-IT and/or @corrideat for another approval before merging.

File: frontend/model/state.js
Comment (Silver-IT) [reply frontend/model/state.js]: Yeah, correct. As we can see in the comment above, we could clear the `state/vuex/postUpgradeVerification` as we have already published version 0.2.0.

**Review by Silver-IT — DISMISSED** [review]
I don't find any issues too.

**Review by corrideat — DISMISSED** [review]

**Review by taoeffect — APPROVED** [review]

**Pull request merged**
